File: back/swift/app/delivery/feature/game/endpoints/GameEndpoints.swift
import Foundation
import Vapor

class GameEndpoints {
    
    private let controller: GameController
    private let router: RoutesBuilder
    
    init(gameController: GameController, router: RoutesBuilder){
        self.controller = gameController
        self.router = router
        initializeRoutes()
    }
    
    private func initializeRoutes() {
        router.post("start") { req -> Response in
            return self.controller.startGame(req)
        }

        router.post("restart") { req -> Response in
            return self.controller.restartGame(req)
        }

        router.post("move") { req -> Response in
            return self.controller.makeMove(req)
        }
    }
    
}

File: back/node/app/delivery/feature/game/endpoints/GameEndpoints.ts
import {GameController} from "../controllers/GameController.js";
import {Router} from "express";

class GameEndpoints {
    
    constructor(
        private controller: GameController,
        private router: Router
    ) {
       this.initializeRoutes()
    }

    initializeRoutes() {
        this.router.get('/hello', (req, res) => this.controller.helloWorld(res));
        this.router.get('/start', (req, res) => this.controller.startGame(res));
        this.router.get('/restart', (req, res) => this.controller.restartGame(req, res));
        this.router.post('/move', (req, res) => this.controller.makeMove(req, res));
    }
    
}

export {GameEndpoints}